include/aspect/volume_of_fluid/utilities.h

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright (C) 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect__volume_of_fluid_utilities_h
+#define _aspect__volume_of_fluid_utilities_h
+
+#include <deal.II/base/point.h>
+
+#include <aspect/global.h>
+
+namespace aspect
+{
+  namespace VolumeOfFluid
+  {
+    using namespace dealii;
+
+    // Utilities
+
+    /**
+     * Function to calculate volume fraction contained by indicator function
+     * H(d-normal*(x'-x_{cen}')) on the [0, 1]^dim unit cell where x_{cen} is
+     * the unit cell center.
+     *
+     * Currently only works assuming constant Jacobian determinant.
+     */
+    template<int dim>
+    double compute_fluid_fraction (const Tensor<1, dim, double> normal,
+                                   const double d);
+
+    /**
+     * Function to calculate required value of d to obtain given volume
+     * fraction for indicator function H(d-normal*(x'-x_{cen}')) on the [0,
+     * 1]^dim unit cell where x_{cen} is the unit cell center.
+     *
+     * Currently only works assuming constant Jacobian determinant.
+     */
+    template<int dim>
+    double compute_interface_location (const Tensor<1, dim, double> normal,
+                                       const double volume_fraction);
+
+    /**
+     * Obtain values at points for a polynomial function that is equivalent to
+     * the Heaviside function $H(d-normal*xhat)$ on the unit cell when
+     * integrated against polynomials of up to the specified degree.
+     *
+     * Currently works for degree <=1
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param d Interface parameter specifying location of interface in unit cell
+     * @param points Locations to evaluate the constructed polynomial
+     * @param values Values of the constructed polynomial at the specified points
+     */
+    template<int dim>
+    void xFEM_Heaviside(const int degree,
```

Comment:
The name of this function does not really explain what it does. What about `Heaviside_equivalent_polynomial_values`? Does this capture the meaning better? If you have a better idea feel free to go for it.